src/main/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervals.java

```diff
@@ -0,0 +1,122 @@
+package org.broadinstitute.hellbender.tools.copynumber;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.File;
+import java.util.List;
+
+
+
+@CommandLineProgramProperties(
+        summary = "Split intervals into sub-interval files.",
+        oneLineSummary = "Split intervals into sub-interval files.",
+        programGroup = VariantProgramGroup.class
+)
+@DocumentedFeature
+public class CreateBinningIntervals extends GATKTool {
+    public static final String WIDTH_OF_BINS_SHORT_NAME = "bw";
+    public static final String WIDTH_OF_BINS_LONG_NAME = "binwidths";
+
+    public static final String PADDING_SHORT_NAME = "pad";
+    public static final String PADDING_LONG_NAME = "padding";
+
+    @Argument(
+            doc = "width of the bins",
+            fullName = WIDTH_OF_BINS_LONG_NAME,
+            shortName = WIDTH_OF_BINS_SHORT_NAME,
+            optional = true,
+            minValue = 1
+    )
+    private int widthOfBins = 1;
+
+    @Argument(
+            doc = "width of the padding regions",
+            fullName = PADDING_LONG_NAME,
+            shortName = PADDING_SHORT_NAME,
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "output file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    public File outputFile;
+
+
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    /**
+     * Generates binning coverage in the intervals given by the user.
+     * The width of bins, the intervals and the output file's path are given by the user.
+     */
+    public void createBins() {
+        // check if the output directory exists
+        if (!outputFile.exists() && !outputFile.mkdir()) {
+            throw new RuntimeException("Unable to create file: " + outputFile.getAbsolutePath());
```

Comment:
Please throw `UserException.CouldNotCreateOutputFile` for such cases. Also, as @samuelklee mentioned, we typically do not create output directories and throw an exception if the output path was not writable. Also, as @samuelklee suggested, please extract _(refactor)_ writing routines out of `createBins()`.  A good place for that is in `doWork()`.

Reply:
As I mentioned above, I think I cannot overwrite doWork() since it is a final method. 
I removed the check for creating the output directory.

Reply:
It can be moved if it is change to be a `CommandLineProgram` instead of `GATKTool`...If you still wanna maintain a `GATKTool`, I guess that the proper method will be `traverse()`